packages/sdk/src/NomadContext.ts

```diff
@@ -174,6 +190,44 @@ export class NomadContext extends MultiProvider<config.Domain> {
     return this.mustGetCore(this.governor.domain);
   }
 
+  /**
+   * Proves and Processes a transaction on the destination chain. This is subsidize and
+   * automatic on non-Ethereum destinations
+   * 
+   * @dev Ensure that a transaction is ready to be processed. You should ensure the following
+   * criteria have been met prior to calling this function:
+   *  1. The tx has been relayed (has status of 2):
+   *       `const { status } = await NomadMessage.events()`
+   *  2. The `confirmAt` timestamp for the tx is in the past:
+   *       `const confirmAt = await NomadMessage.confirmAt()`
+   *
+   * @param message NomadMessage
+   * @returns The Contract Transaction receipt
+   */
+  async process(message: NomadMessage<NomadContext>): Promise<ContractTransaction>{
+    const s3URL = `https://nomadxyz-${this.environment}-proofs.s3.us-west-2.amazonaws.com/`;
+
+    const originNetwork = this.resolveDomainName(message.origin);
+    const destNetwork = this.resolveDomainName(message.destination);
+    const index = message.leafIndex.toString();
+    const s3Res = await fetch(`${s3URL}${originNetwork}_${index}`);
+    if (!s3Res) throw new Error('Not able to fetch proof');
+    const data: MessageProof = await s3Res.json();
+
+    // get replica contract
+    const replica = this.mustGetReplicaFor(originNetwork, destNetwork);
+
+    // get signer and connect replica
+    const signer = this.mustGetSigner(destNetwork);
```

Comment:
I think this is unnecessary? as `getReplica` is the underlying implementation here, and that does connect

https://github.com/nomad-xyz/monorepo/blob/f697b3649fc179dc3391fe1b0f68cb26062c6fad/packages/new-deploy/src/core/CoreContracts.ts#L127-L137

Reply:
Cool! Yeah, that works, just tested it out